org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/BaseLaunchRequestJob.java

```diff
@@ -70,12 +72,23 @@ protected final void runToolingApiJob(IProgressMonitor monitor) {
         // fetch build environment
         List<ProgressListener> listeners = ImmutableList.<ProgressListener>of(DelegatingProgressListener.withFullOutput(monitor));
 
-        // apply the fixed attributes on the request o
         BuildRequest<Void> request = createRequest();
-        FixedRequestAttributes fixedAttributes = getConfigurationAttributes().toFixedRequestAttributes();
-        fixedAttributes.apply(request);
-
-        // configure the request's transient attributes
+        RunConfiguration runConfig = getRunConfig();
+        BuildConfiguration buildConfig = runConfig.getBuildConfiguration();
+        // TODO (donat) this won't be necessary once the project config can override Gradle user home
+        WorkspaceConfiguration workspaceConfig = CorePlugin.configurationManager().loadWorkspaceConfiguration();
+
+        request.gradleDistribution(buildConfig.getGradleDistribution());
+
+        // apply the fixed attributes
+        request.projectDir(buildConfig.getRootProjectDirectory());
+        request.gradleUserHomeDir(workspaceConfig.getGradleUserHome());
```

Comment:
The build config should have a reference to the workspace config, so we don't need to load it separately here.